engine/src/main/scala/cromwell/engine/io/nio/NioHashing.scala

```diff
@@ -74,13 +71,63 @@ object NioHashing {
     }
   }
 
-  private def getFileHashForGcsPath(gcsPath: GcsPath): IO[FileHash] = delayedIoFromTry {
-    gcsPath.objectBlobId.map(id => FileHash(HashType.GcsCrc32c, gcsPath.cloudStorage.get(id).getCrc32c))
-  }
+  private def getFileHashForGcsPath(gcsPath: GcsPath, hashStrategy: FileHashStrategy): IO[Option[FileHash]] =
```

Comment:
Also extremely optional, I wonder if we could apply ✨ More Types ✨ to DRY out this group of functions.

```suggestion
  private def getFileHashForPath[T](path: T, hashStrategy: FileHashStrategy[T]): IO[Option[FileHash]] =
```

Reply:
We could, it would just mean that function would have a big match/case statement in it... which might be nicer, I'll give it a try. I kind of like the idea of doing cool stuff with `FileHashStrategy[T]` but don't want to hold this up that much further.

Reply:
Discussed offline and decided to leave this as-is.